plugins/orchestrator-backend/src/service/router.ts

```diff
@@ -297,15 +303,12 @@ function setupInternalRoutes(
       if (decision.result === AuthorizeResult.DENY) {
         manageDenyAuthorization(endpointName, endpoint, req);
       }
-      await routerApi.v2
+      return routerApi.v2
```

Comment:
but getWorkflowsOverview is `async` is it ok to return a Promise?

Reply:
yes

Reply:
it shouldn't be with .then and .catch, but I didn't have time to change that

Reply:
the way it should be with an async function is:
try {
  const result = await getWorkflowsOverview(buildPagination(req), getRequestFilters(req))
  res.json(result)
}  catch (err) {
    auditLogRequestError(error, endpointName, endpoint, req);
     next(error);
}

these router functions combine classic promise coding (with .then and .catch) with async await. This is bad practice.

but removing the await is the least of the evil, at least this line of code doesn't combine the two.

Reply:
ok, please add a jira item the improvement epic to track this.

Reply:
done
https://issues.redhat.com/browse/FLPATH-1844